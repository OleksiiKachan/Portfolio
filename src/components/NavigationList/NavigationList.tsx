import React, { Component } from 'react';
import classnames from 'classnames';

/*
    <NavigationList navigationItems=[] onLinkClick={}/>
*/
class NavigationList extends Component<any, any> {
  render() {
    return (
      <div className={classnames('navigation__listWrapper')}>
        <ul className={classnames('navigation__list')}>
          {this.props.navigationItems.map(
            (navigationItem: {
              key: string;
              sectionId: string;
              title: string;
            }) => {
              return (
                <li
                  className={classnames('navigation__listItem')}
                  key={navigationItem.key}
                >
                  <a
                    onClick={this.props.onLinkClick}
                    href={navigationItem.sectionId}
                    className={classnames('navigation__listLink')}
                  >
                    {navigationItem.title}
                  </a>
                </li>
              );
            }
          )}
        </ul>
      </div>
    );
  }
}

export default NavigationList;
