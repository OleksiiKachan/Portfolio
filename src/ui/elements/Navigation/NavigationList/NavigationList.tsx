import React, { Component } from 'react';
import classnames from 'classnames';
import { Link } from 'react-router-dom';
import { NavigationItemType } from '../../../../lib/types';

/*
    <NavigationList
      navigationItems=[]
      onLinkClick={}
      className=''
    />
*/

type PropsType = {
  navigationItems: Array<NavigationItemType>;
  onLinkClick: (selectedItemKey: string) => void;
  className?: string;
};

export default class NavigationList extends Component<PropsType, any> {
  render() {
    return (
      <div
        className={classnames('navigation__listWrapper', this.props.className)}
      >
        <ul className={classnames('navigation__list')}>
          {this.props.navigationItems.map(
            (navigationItem: {
              key: string;
              sectionUrl: string;
              title: string;
              isSelected: boolean;
            }) => {
              return (
                <li
                  className={classnames(
                    'navigation__listItem',
                    navigationItem.isSelected && 'navigation__listItem_selected'
                  )}
                  key={navigationItem.key}
                >
                  <Link
                    to={navigationItem.sectionUrl}
                    onClick={() => {
                      this.props.onLinkClick(navigationItem.key);
                    }}
                    className={classnames('navigation__listLink')}
                  >
                    {navigationItem.title}
                  </Link>
                </li>
              );
            }
          )}
        </ul>
      </div>
    );
  }
}
