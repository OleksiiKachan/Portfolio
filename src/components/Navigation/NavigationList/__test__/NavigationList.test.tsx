import React from 'react';
import { shallow } from 'enzyme';
import NavigationList from '..';

describe('NavigationList', () => {
  it('Snapshot', () => {
    const component = shallow(
      <NavigationList
        navigationItems={[
          {
            key: 'mi_education',
            title: 'Education & Experience',
            sectionUrl: '/experience',
            isSelected: false,
          },
          {
            key: 'mi_contact',
            title: 'Contact',
            sectionUrl: '/contact',
            isSelected: false,
          },
        ]}
        onLinkClick={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
