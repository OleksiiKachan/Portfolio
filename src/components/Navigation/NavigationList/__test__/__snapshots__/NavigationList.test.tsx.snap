// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationList Snapshot 1`] = `
<div
  className="navigation__listWrapper"
>
  <ul
    className="navigation__list"
  >
    <li
      className="navigation__listItem"
      key="mi_education"
    >
      <Link
        className="navigation__listLink"
        onClick={[Function]}
        to="/experience"
      >
        Education & Experience
      </Link>
    </li>
    <li
      className="navigation__listItem"
      key="mi_contact"
    >
      <Link
        className="navigation__listLink"
        onClick={[Function]}
        to="/contact"
      >
        Contact
      </Link>
    </li>
  </ul>
</div>
`;
